program/src/switchboard_set_weight.rs

```diff
@@ -0,0 +1,93 @@
+use jito_bytemuck::AccountDeserialize;
+use jito_restaking_core::ncn::Ncn;
+use jito_tip_router_core::{
+    constants::{SWITCHBOARD_MAX_STALE_SLOTS, WEIGHT_PRECISION},
+    error::TipRouterError,
+    weight_table::WeightTable,
+};
+use solana_program::{
+    account_info::AccountInfo, clock::Clock, entrypoint::ProgramResult, msg,
+    program_error::ProgramError, pubkey::Pubkey, sysvar::Sysvar,
+};
+use switchboard_on_demand::{
+    prelude::rust_decimal::{prelude::ToPrimitive, Decimal},
+    PullFeedAccountData,
+};
+
+/// Updates weight table
+pub fn process_switchboard_set_weight(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    st_mint: Pubkey,
+    epoch: u64,
+) -> ProgramResult {
+    let [ncn, weight_table, switchboard_feed] = accounts else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    Ncn::load(&jito_restaking_program::id(), ncn, false)?;
+
+    WeightTable::load(program_id, weight_table, ncn, epoch, true)?;
+
+    let (registered_switchboard_feed, no_feed_weight) = {
+        let weight_table_data = weight_table.data.borrow();
+        let weight_table_account = WeightTable::try_from_slice_unchecked(&weight_table_data)?;
+
+        let weight_entry = weight_table_account.get_weight_entry(&st_mint)?;
+
+        (
+            weight_entry.mint_entry().switchboard_feed(),
+            weight_entry.mint_entry().no_feed_weight(),
+        )
+    };
+
+    let weight: u128 = if registered_switchboard_feed.eq(&Pubkey::default()) {
+        no_feed_weight
```

Comment:
maybe a msg! here that we are going with fallback weight. a bit worried about scenarios where fallback weight is being used when we thought there was a feed or forgot to add a feed